INSTALL

```diff
@@ -0,0 +1,7 @@
+To install from source
+
+cp Makeconf.local.template Makeconf.local
+
```

Comment:
I don't see Makeconf.local.template. I think you forgot to add it to the git. 

This is a good idea in my opinion. I would say we move current config from makeconf into makefile and rename Makeconf into Makeconf.template (with adjustments). Then people could do `cp Makeconf.template Makeconf`. How does this sound?

Reply:
No, sorry!  `Makeconf` is crucial in the current setup, and for normal users, including me, for *Release*  it is both necessary and sufficient.

Please do not (again!) break the  'make' seetup  by incorrect cleanup (!).
This was particularly bad, for 18.10 and notably broke things for the downstream packaging people.

Reply:
@mmaechler, Makeconf is tracked by git, and that's bad setup for a custom configuration file. What @dickmao proposes is superior to what we have right now. You will still have your Makeconf as Makeconf.template.

Reply:
Can't we just define all the variables in Makeconf with `=?` and then put `-include config.mk` before `include Makeconf` in all the various makefiles? That seems much cleaner to me.

Reply:
The order is currently `include Makeconf`, and then `include Makeconf.local`.  I would avoid `?=` if it's unnecessary (which is why I took most of them out in 17b99cf.  I also don't know why people cannot see `Makeconf.local.template` -- it seems to be git added.

Reply:
> avoid ?= if it's unnecessary 

Why? Using it how I suggested lets users override Makeconf variables if they want and by using `-include` we don't have to do all the checks you do for whether or not the file exists.

Reply:
We can either do as you propose which is replacing all assignments in `Makeconf` with `?=` and include `Makeconf.local` *before* `Makeconf`, ... or....

We can leave `Makeconf` alone, and include `Makeconf.local` *after* `Makeconf`.  Perhaps I am missing something.

Reply:
and include Makeconf.local before Makeconf

`-include`, not `include`. It simplifies this:

```
LOCAL_CONF := $(strip $(wildcard ./Makeconf.local))
ifeq ($(LOCAL_CONF),)
$(error "No ./Makeconf.local found.  Please read INSTALL")
else
include ./Makeconf.local
endif
```

to just 

```
-include Makeconf.local
include Makeconf
```

Reply:
I would prefer the former (despite the fact that removing lines is (almost) always better than adding lines).  Lest we get ahead of ourselves, I am in favor of suspending discussion over this minutiae until as such time I can test the "rules of engagement" for debian and/or rpm.

Reply:
The former throws and that will confuse a lot of users who are used to `make install` just working out of the box.

How about splitting this in 2 parts, package-build test target and makeconf.local + install? We pull the non-controversial package-build  and have a separate discussion on the other part.